workflow/installertest.groovy

```diff
@@ -0,0 +1,191 @@
+ *              Anything else will be downloaded via wget
+ */
+def fetchArtifact(String url) {
+  if (url == null) {
+    fail 'required parameter url is missing'
+  } else if (url.startsWith("artifact://")) {
+    echo "Fetching ${url} as artifact."
+    def comp = this.getComponentsFromArtifactUrl(url)
+    step([$class: 'CopyArtifact', filter: comp.artifact, projectName: comp.item, flatten: true, selector: [$class: 'SpecificBuildSelector', buildNumber: comp.run]])
+  } else {
+      echo "Fetching ${url} as URL file."
+      sh "wget -q ${url}"
+  }
+}
+
+// Pull down the artifacts, must run in a node block
+def fetchInstallers(String debFileUrl, String rpmFileUrl, String suseFileUrl) {
+ sh 'rm -rf installers'
+ dir('installers') {
+   sh 'rm -rf deb rpm suse'
+   dir('deb') {
+      fetchArtifact(debFileUrl)
+   }
+   dir('rpm') {
+      fetchArtifact(rpmFileUrl)
+   }
+   dir('suse') {
+      fetchArtifact(suseFileUrl)
+   }
+  }
+}
+
+/** Runs a series of shell commands inside a docker image
+* The output of each command set is saved to a specific file and archived
+* Errors are propagated back up the chain. 
+* @param imageName docker image name to use (must support sudo, based off the sudoable images)
+* @param shellCommands List of (string) shell commands to run within the container
+* @param stepNames List of names for each shell command step, optional
+*                    (if not supplied, then the step # will be used)
+*/
+def runShellTest(String imageName, def shellCommands, def stepNames=null) {
+  withEnv(['HOME='+pwd()]) { // Works around issues not being able to find docker install
+    def img = docker.image(imageName)
+    def fileName = convertImageNameToString(imageName,"-testOutput-")
+    img.inside() {  // Needs to be root for installation to work
+      try {
+        for(int i=0; i<shellCommands.size(); i++) {
+          String cmd = shellCommands.get(i)
+          def name = (stepNames != null && i < stepNames.size()) ? stepNames.get(i) : i
+          
+          // Workaround for two separate and painful issues
+          // One, in shells, piped commands return the exit status of the last command
+          // This means that errors in our actual command get eaten by the success of the tee we use to log
+          // Thus, failures would get eaten and ignored. 
+          // Setting pipefail in bash fixes this by returning the first nonsuccessful exit in the pipe
+
+          // Second, the sh workflow step often will use the default posix shell
+          // The default posix shell does not support pipefail, so we have to invoke bash to get it
+          
+          String argument = 'bash -c \'set -o pipefail; '+cmd+" | tee \"testresults/$fileName-$name"+'.log'+'" \''
+          sh argument
+        }
+      } catch (Exception ex) {
+        archive("testresults/$fileName"+'*.log')
+        throw ex
+      }
+      archive("testresults/$fileName"+'*.log')
```

Comment:
I had some quirky behaviors in testing with this (nothing that I could call an out-and-out bug, but quirky).  So I left it with something I knew worked correctly.